lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -6354,6 +6354,16 @@ namespace Js
         return this->propertyStringMap;
     }
 
+    EnumeratorCache* JavascriptLibrary::GetObjectAssignCache(Type* type)
+    {
+        if (this->cache.assignCache == nullptr)
+        {
+            this->cache.assignCache = AllocatorNewArrayZ(CacheAllocator, scriptContext->GetEnumeratorAllocator(), EnumeratorCache, Cache::AssignCacheSize);
+        }
+
+        return &this->cache.assignCache[(((size_t)type) >> PolymorphicInlineCacheShift) & (Cache::AssignCacheSize - 1)];
```

Comment:
>AssignCacheSize [](start = 98, length = 15)

Maybe we could add a static assertion that AssignCacheSize is a power of 2, to avoid surprising out-of-bounds behavior if someone tries tweaking the constant.